AstSemantics.md

```diff
@@ -32,6 +32,18 @@ Callstack space is limited by unspecified and dynamically varying constraints
 and is a source of [nondeterminism](Nondeterminism.md). If program callstack usage
 exceeds the available callstack space at any time, a trap occurs.
 
+Implementations are not permitted to do implicit tail-call optimizations.
+Semantically, implementations must have an internal maximum call stack size,
+and every call must take up some resources toward exhausting that size. Of
+course, dynamic resources may be exhausted much earlier. This rule exists
+to avoid programs depending on tail call optimizations which may not be
+portable across implementations. Also, in the future, it is expected that
+WebAssembly will add some form of stack-introspection functionality, in which
+case such optimizations would be directly observable.
+
+Support for explicit tail calls is planned in
+[the future](FutureFeatures.md#general-purpose-proper-tail-calls).
```

Comment:
To address the above objection, perhaps append "which would add an explicit tail-call operation with well-defined effects on stack introspection."?